src/structureUtilities.jl

```diff
@@ -10,15 +10,134 @@ Arguments:
 * `algo`: Algorithm, one out of [:learnspn, :random].
 """
 function generate_spn(X::Matrix, algo::Symbol; params...)
-
     if algo == :learnspn
         return learnspn(X; params...)
     elseif algo == :random
         return randomspn(X; params...)
     else
-        @error("Unknown structure learning method: ", method)
+        @error("Unknown structure learning algorithm: ", algo)
+    end
+end
+
+"""
+    learnspn(X; distribution=Normal(), minclustersize=100)
+
+Return Sum Product Network learned by a simplified LearnSPN algorithm.
+"""
+function learnspn(X; distribution=Normal(), minclustersize=100)
+    q = Queue{Tuple}()
+    root = FiniteSumNode()
+    instances = collect(1:size(X)[1])
+    variables = collect(1:size(X)[2])
+    enqueue!(q, (root, variables, instances))
+    
+    while length(q) > 0
+        node, variables, instances = dequeue!(q)
+
+        # stopping condition, one variable left, estimate distribution
+        if length(variables) == 1
+            v = variables[1]
+            slice = X[instances, v]
+            add!(node, UnivariateNode(mle(distribution, X[:, v]), v))
```

Comment:
You can replace this with the `fit` function of the `Distributions` package.

```Julia
add!(node, UnivariateNode(fit(distributions[v], X[instances,v]), v))
```

Btw.: your current code here is buggy because you don't use slice for the fitting.